link errors when linking against fftw

fftw installs two groups of libraries: plain and `_threads`:

Threadless:
  * libfftw3.a
  * libfftw3f.a
  * libfftw3l.a

Threadful:
  * libfftw3_threads.a
  * libfftw3f_threads.a
  * libfftw3l_threads.a

File `fftw*.pc` doesn't include `-lfftw3_threads`:
```pc
prefix=/home/mxe/mxe-build-pkg-args/usr/i686-w64-mingw32.static
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: FFTW
Description: fast Fourier transform library
Version: 3.3.4
Libs: -L${libdir} -lfftw3 
Libs.private: -lm
Cflags: -I${includedir}
```

File `fftw.pc.in` from upstream doesn't include `-lfftw3_threads`:
```pc
prefix=@prefix@
exec_prefix=@exec_prefix@
libdir=@libdir@
includedir=@includedir@

Name: FFTW
Description: fast Fourier transform library
Version: @VERSION@
Libs: -L${libdir} -lfftw3@PREC_SUFFIX@ @LIBQUADMATH@
Libs.private: -lm
Cflags: -I${includedir}
```

`PREC_SUFFIX` is one letter suffix for precision ('f', 'd', etc). `LIBQUADMATH` is bout linking with `quadmath`.

Linker errors:
```
i686-w64-mingw32.static-g++ -o mcf_levelsets3d.exe mcf_levelsets3d.cpp -I.. -Wall -W -O2     -Dcimg_use_tiff   -Dcimg_use_openexr -I/home/mxe/mxe-build-pkg-args/usr/i686-w64-mingw32.static/include/OpenEXR   -Dcimg_use_png -I/home/mxe/mxe-build-pkg-args/usr/i686-w64-mingw32.static/include/libpng16   -Dcimg_use_jpeg   -Dcimg_use_zlib    -Dcimg_use_opencv -I/home/mxe/mxe-build-pkg-args/usr/i686-w64-mingw32.static/include/opencv -I/home/mxe/mxe-build-pkg-args/usr/i686-w64-mingw32.static/include/OpenEXR   -Dcimg_use_magick -I/home/mxe/mxe-build-pkg-args/usr/i686-w64-mingw32.static/include/ImageMagick-6 -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -D_LIB   -Dcimg_use_fftw3   -lm -ltiff -llzma -ljpeg -lz   -ltiffxx -lIlmImf -lImath -lHalf -lIex -lIexMath -lIlmThread -lz   -lpng16 -lz   -ljpeg   -lz    -lopencv_contrib -lopencv_stitching -lopencv_nonfree -lopencv_superres -lopencv_ocl -lopencv_ts -lopencv_videostab -lopencv_gpu -lopencv_photo -lopencv_objdetect -lopencv_legacy -lopencv_video -lopencv_ml -lopencv_calib3d -lopencv_features2d -lopencv_highgui -lopencv_imgproc -lopencv_flann -lopencv_core -lstdc++ -lglu32 -lopengl32 -lpng -ljasper -lcomctl32 -lgdi32 -lole32 -lsetupapi -lws2_32 -lstrmiids -lavifil32 -lavicap32 -lwinmm -lmsvfw32 -lIlmImf -lImath -lHalf -lIex -lIexMath -lIlmThread -ltiff -luuid -loleaut32 -llzma -ljpeg -lz   -lMagick++-6.Q16 -lMagickWand-6.Q16 -lMagickCore-6.Q16 -llcms2 -ljpeg -llqr-1 -lfftw3 -lwebp -llzma -lIlmImf -lImath -lHalf -lIex -lIexMath -lIlmThread -lpangocairo-1.0 -lpangoft2-1.0 -lpangowin32-1.0 -lpango-1.0 -lharfbuzz -lcairo -lgobject-2.0 -lfontconfig -lmsimg32 -lpixman-1 -lexpat -lfreetype -lbz2 -lpng16 -lharfbuzz_too -lglib-2.0 -lusp10 -L/home/mxe/mxe-build-pkg-args/usr/i686-w64-mingw32.static/lib/../lib -lffi -lws2_32 -lole32 -lwinmm -lshlwapi -lpcre -lintl -liconv -lz -lgdi32 -lm   -lfftw3 -lm  
/tmp/cc7GCZpf.o:CImg_demo.cpp:(.text$_ZN12cimg_library4CImgIfE3FFTERS1_S2_bj[__ZN12cimg_library4CImgIfE3FFTERS1_S2_bj]+0x2bd): undefined reference to `fftw_plan_with_nthreads'
/tmp/cc7GCZpf.o:CImg_demo.cpp:(.text$_ZN12cimg_library4CImgIfE3FFTERS1_S2_bj[__ZN12cimg_library4CImgIfE3FFTERS1_S2_bj]+0x6eb): undefined reference to `fftw_cleanup_threads'
/tmp/cc7GCZpf.o:CImg_demo.cpp:(.text$_ZN12cimg_library4CImgIfE3FFTERS1_S2_bj[__ZN12cimg_library4CImgIfE3FFTERS1_S2_bj]+0x9b7): undefined reference to `fftw_init_threads'
collect2: error: ld returned 1 exit status
```

FFTW's site [says](http://www.fftw.org/doc/Usage-of-Multi_002dthreaded-FFTW.html) that for threaded usage linker flags should be `-lfftw3_threads -lfftw3 -lm`. (If openmp is not used. In MXE, fftw doesn't depend on openmp, so I think openmp is not used.)

I propose to add `-lfftw3_threads` to `fftw*.pc` files.

--- a/src/fftw.mk
+++ b/src/fftw.mk
@@ -21,13 +21,15 @@ endef
 define $(PKG)_BUILD
     cd '$(1)' && ./configure \
         $(MXE_CONFIGURE_OPTS) \
-        --enable-threads
+        --enable-threads \
+        --with-combined-threads
     $(MAKE) -C '$(1)' -j '$(JOBS)' bin_PROGRAMS= sbin_PROGRAMS= noinst_PROGRAMS=
     $(MAKE) -C '$(1)' -j 1 install bin_PROGRAMS= sbin_PROGRAMS= noinst_PROGRAMS=
 
     cd '$(1)' && ./configure \
         $(MXE_CONFIGURE_OPTS) \
         --enable-threads \
+        --with-combined-threads \
         --enable-long-double
     $(MAKE) -C '$(1)' -j '$(JOBS)' bin_PROGRAMS= sbin_PROGRAMS= noinst_PROGRAMS=
     $(MAKE) -C '$(1)' -j 1 install bin_PROGRAMS= sbin_PROGRAMS= noinst_PROGRAMS=
@@ -35,6 +37,7 @@ define $(PKG)_BUILD
     cd '$(1)' && ./configure \
         $(MXE_CONFIGURE_OPTS) \
         --enable-threads \
+        --with-combined-threads \
         --enable-float
     $(MAKE) -C '$(1)' -j '$(JOBS)' bin_PROGRAMS= sbin_PROGRAMS= noinst_PROGRAMS=
     $(MAKE) -C '$(1)' -j 1 install bin_PROGRAMS= sbin_PROGRAMS= noinst_PROGRAMS=

